copilot-verifier/src/Copilot/Verifier.hs

```diff
@@ -599,8 +600,10 @@ copilotExprToRegValue sym = loop
       pure V.empty
     loop (CW4.XArray xs) (VectorRepr tpr) =
       V.generateM (PVec.lengthInt xs) (\i -> loop (PVec.elemAtUnsafe i xs) tpr)
-
-    loop CW4.XStruct{} _ = fail "FIXME: struct"
+    loop (CW4.XStruct xs) (StructRepr ctx) =
+      Ctx.traverseWithIndex
+        (\i tpr -> RV <$> loop (xs !! Ctx.indexVal i) tpr)
```

Comment:
The use of `(!!)` here is a bit ugly, but I'm not sure if we can do better with the way the data structures involved are currently defined. (In fact, `copilot-theorem` itself does the same thing [here](https://github.com/GaloisInc/copilot-1/blob/ebd9b227eb610c02ab719197d469dd66edcc869c/copilot-theorem/src/Copilot/Theorem/What4/Translate.hs#L564).) We could reduce the asymptotic complexity by changing the field of `XStruct` from a list to a `Vector`, although I don't know if this is actually a performance bottleneck in practice.

Reply:
I'd be shocked if list vs vector made any perceptible difference here.  My experience with vectors is that they often are actually worse unless constructed all-at-once because they do so much copying.